dpm/client/__init__.py

```diff
@@ -149,6 +154,43 @@ def publish(self, publisher=None):
         # Return published datapackage url
         return self.server + '/%s/%s' % (self.username, self.datapackage.descriptor['name'])
 
+    def _prepare_files_authorize(self):
+        files = filter(isfile, listdir(self.datapackage.base_path))
+
+        accepted_descriptor = 'datapackage.json'
+        descriptor = [f for f in files if f == accepted_descriptor][0]
+        file_list = [descriptor]
+        for resource in self.datapackage.resources:
```

Comment:
Blank line before this line would make this easier to read. I'd also suggest we have README before resources in the upload sequence.